services/delivery/src/deliver/get-ad.js

```diff
@@ -39,7 +37,15 @@ const updateCorrelator = async (correlator, ad, lineitemId) => {
  */
 module.exports = async (correlator, adunit, date) => {
   const correlated = await db.findOne('correlators', { value: correlator });
-  if (correlated) return correlated;
+  if (correlated) {
+    // Return the correlated item's ad src/url
+    const ad = await db.strictFindActiveById('ads', correlated.adId, { projection: { 'image.src': 1, url: 1 } });
```

Comment:
I believe this should be a normal find by ID, not strict and active, since those rules don't seem to be applied to correlators